chanzuckerberg/corpora-data-portal
# Thuang 518 download part 2

1. Rebased from Trent's Part 1 PR!

1. Add two smoke tests to assert clicking on the "download" button in the column triggers the modal, and also assert the user flow of selecting a data format and the download link exists 

1. Update `backend/config/corpora-api.yml` to match actual response keys

1. Remove temporary fixture: `frontend/src/common/fixtures/projects.ts`, since now we have local database and dev API endpoint to use

1. Add functionality for download modal - will comment in code for clarity! 

![gif](https://user-images.githubusercontent.com/6309723/93391255-bc01cd80-f823-11ea-95e5-242dcec0c159.gif)

QA Steps:
1. Make sure your `frontend/src/configs/configs.js` points to `API_URL: "https://api.dev.corpora.cziscience.com",`, so you have more datasets to test
2. Start FE server! `npm run develop`
3. Click on any of the "Download" button in the Dataset rows, and it should trigger a modal
4. Select a data format
5. Either copy CURL and run the command in the terminal OR click "Download" to start downloading
6. That's it :D

Thank you!!
